src/asset_list.rs

```diff
@@ -87,14 +89,45 @@ impl AssetList {
 
     /// Return a copy of the underlying vector
     pub fn to_vec(&self) -> Vec<Asset> {
-        self.0.clone()
+        self.0.to_vec()
     }
 
     /// Return length of the asset list
     pub fn len(&self) -> usize {
         self.0.len()
     }
 
+    pub fn iter(&self) -> Iter<Asset> {
+        self.0.iter()
+    }
+
+    pub fn iter_mut(&mut self) -> IterMut<Asset> {
+        self.0.iter_mut()
+    }
```

Comment:
ideally would like to implement the relavent traits instead of this but couldnt get that to work, will look into it more tm

Reply:
It seems like these are actually not traits, and are direct impls on e.g. Vec.

Reply:
i think this is fine as is since the corresponding impls on vec are not trait impls